src/lib/Zikula/Core/Helper/ExtensionVar.php

```diff
@@ -0,0 +1,283 @@
+     * @todo consider \Symfony\Component\HttpFoundation\ParameterBag
+     */
+    private $variables;
+
+    /**
+     * ExtensionVar constructor.
+     * @param EntityManagerInterface $entityManager
+     * @param KernelInterface $kernel
+     */
+    public function __construct(EntityManagerInterface $entityManager, KernelInterface $kernel)
+    {
+        $this->entityManager = $entityManager;
+        $this->kernel = $kernel;
+        $this->initExtensionVars();
+    }
+
+    /**
+     * Preloads extension vars for a number of key modules to reduce sql statements.
+     *
+     * @param boolean $force
+     *
+     * @return void
+     */
+    public function initExtensionVars($force = false)
+    {
+        // The empty arrays for handlers and settings are required to prevent messages with E_ALL error reporting
+        $this->variables = new \ArrayObject(array(
+            \EventUtil::HANDLERS => array(),
+            \ServiceUtil::HANDLERS => array(),
+            'ZikulaSettingsModule' => array(),
+        ));
+
+        // don't init vars during the installer or upgrader
+        if (!$force && \System::isInstalling()) {
+            return;
+        }
+
+        // Load all variables into the variables property.
+        /** @var ExtensionVarEntity[] $vars */
+        $vars = $this->entityManager->getRepository('Zikula\Core\Doctrine\Entity\ExtensionVarEntity')->findAll();
+        foreach ($vars as $var) {
+            if (!array_key_exists($var->getModname(), $this->variables)) {
+                $this->variables[$var->getModname()] = array();
+            }
+            if (array_key_exists($var->getName(), $GLOBALS['ZConfig']['System'])) {
+                $this->variables[$var->getModname()][$var->getName()] = $GLOBALS['ZConfig']['System'][$var->getName()];
+            } else {
+                $this->variables[$var->getModname()][$var->getName()] = $var->getValue();
+            }
+        }
+
+        // Pre-load the variables array with empty arrays for known extensions that do not define any variables to
+        // prevent unnecessary queries to the database.
+        foreach ($this->kernel->getBundles() as $bundle) {
+            if (!array_key_exists($bundle->getName(), $this->variables)) {
+                $this->variables[$bundle->getName()] = array();
+            }
+        }
+    }
+
+    /**
+     * Checks to see if an extension variable is set.
+     *
+     * @param string $extensionName The name of the extension.
+     * @param string $variableName The name of the variable.
+     *
+     * @return boolean True if the variable exists in the database, false if not.
+     */
+    public function has($extensionName, $variableName)
+    {
+        // define input, all numbers and booleans to strings
+        $extensionName = isset($extensionName) ? ((string) $extensionName) : '';
+        $variableName = isset($variableName) ? ((string) $variableName) : '';
```

Comment:
I guess to ensure other types don't get passed to the DB. you cannot enforce `(string)` args in a function until php 7 so this is the best way for now. to rely on the end developer doing it right is probably not wise.